zigConsole/.vscode/tasks.json

```
@@ -0,0 +1,190 @@
+{
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "build-debug-amd64",
+            "detail": "Builds the project for the remote amd64 target",
+            "command": "zig",
+            "type": "shell",
+            "args": [
+                "build"
+            ],
+            "problemMatcher": [
+                {
+                    "owner": "zig",
+                    "fileLocation": ["relative", "${workspaceRoot}"],
+                    "pattern": {
+                        "regexp": "^(.*):(\\d+):(\\d+):\\s+(\\d+):(\\d+)\\s+(warning|error):\\s+(.*)$",
+                        "file": 1,
+                        "line": 2,
+                        "column": 3,
+                        "endLine": 4,
+                        "endColumn": 5,
+                        "severity": 6,
+                        "message": 7
+                    }
+                }
+            ],
+            "icon": {
+                "id": "flame",
+                "color": "terminal.ansiYellow"
+            },
+            "dependsOrder": "sequence"
+        },
+        {
+            "label": "build-debug-arm64",
+            "detail": "Builds the project for the remote arm64 target",
+            "command": "DOCKER_HOST=",
+            "type": "shell",
+            "args": [
+                "docker",
+                "run",
+                "--rm",
+                "-it",
+                "-v",
+                "${workspaceFolder}:/app",
+                "cross-toolchain-arm64-__container__",
+                "zig",
+                "build",
```

since all the builds will be stored in the same zig-cache and zig-out I think that we can add here an argument:

```suggestion
                "build",
                "--cache-dir", "zig-aarch64",
```

so, we can maintain the caches in different folders and reuse

---

please add this for all the "build-debug-<arch>"

---

Regardless of the zig-cache, it is important to note that it still needs the global-cache in home/.cache/zig`

```bash
 --cache-dir [path]           Override path to local Zig cache directory # (only your application)
  --global-cache-dir [path]    Override path to global Zig cache directory # (build - libcompiler-rt, libc, zig-pkg , libcxx, libunwind)
```

---

> please add this for all the "build-debug-"

Do you want me to separate the binaries copied to the zig-out?

For searching in the zig-cache it will depend on the hash generated. This would mean that if the build is done several times under the same target with some changes, zig-cache will retain the old objects and add new objects and final binaries. However, users don't know exactly which hash was generated by the latest build.

---

The original proposal had been to add a custom path separating the final binaries copied from the `zig-cache`.
However, it was reversed since I had an issue with docker where the binary wasn't recognized nor found by the path specified.
https://github.com/toradex/vscode-torizon-templates/blob/08eaf344e7f3145138daf22eaa7270fd0b16a661/zigConsole/Dockerfile#L103-L104

```diff
diff --git a/zigConsole/build.zig b/zigConsole/build.zig
index c7a870e..b538f2d 100644
--- a/zigConsole/build.zig
+++ b/zigConsole/build.zig
@@ -32,4 +32,30 @@ pub fn build(b: *std.Build) void {
 
     // copy binary from zig-cache to zig-out/bin [default]
     b.installArtifact(binary);
+
+    // overwrite default output
+    b.resolveInstallPrefix(b.fmt("zig-deploy/{s}/{s}", .{ target.linuxTriple(b.allocator) catch unreachable, @tagName(optimize) }), .{});
```

---

### Suggestion

Change:
```zig
    // overwrite default output
    b.resolveInstallPrefix(b.fmt("zig-out/{s}/{s}", .{ @tagName(target.getCpuArch()), @tagName(optimize) }), .{});
    // tagName convert Enum fields to string ([]const u8)
```

**Output:**
```bash
zig-out
├── aarch64
│   ├── Debug
│   │   └── bin
│   │       └── __change__
│   └── ReleaseSafe
│       └── bin
│           └── __change__
└── x86_64
    ├── ReleaseFast
    │   └── bin
    │       └── __change__
    └── ReleaseSafe
        └── bin
            └── __change__
```

---

sounds good to me, go ahead!

---

ok. New commit